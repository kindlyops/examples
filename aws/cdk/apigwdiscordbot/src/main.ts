import * as path from 'path';
import { Runtime } from '@aws-cdk/aws-lambda';
import { NodejsFunction } from '@aws-cdk/aws-lambda-nodejs';
import { App, Construct, Duration, Stack, StackProps } from '@aws-cdk/core';
import { DiscordBotConstruct } from 'discord-bot-cdk-construct';


export class SampleDiscordBotStack extends Stack {
  /**
   * The constructor for building the stack.
   * @param {Construct} scope The Construct scope to create the stack in.
   * @param {string} id The ID of the stack to use.
   */
  constructor(scope: Construct, id: string, props: StackProps = {}) {
    super(scope, id, props);

    // Create the Commands Lambda.
    const discordCommandsLambda = new NodejsFunction(this, 'discord-commands-lambda', {
      runtime: Runtime.NODEJS_14_X,
      entry: path.join(__dirname, './functions/commands.ts'),
      handler: 'handler',
      timeout: Duration.seconds(60),
    });

    new DiscordBotConstruct(this, 'discord-bot-endpoint', {
      commandsLambdaFunction: discordCommandsLambda,
    });
  }
}

// for development, use account/region from cdk cli
const devEnv = {
  account: process.env.CDK_DEFAULT_ACCOUNT,
  region: process.env.CDK_DEFAULT_REGION,
};

const app = new App();

new SampleDiscordBotStack(app, 'discord-hello-bot', { env: devEnv });
// new MyStack(app, 'my-stack-prod', { env: prodEnv });

app.synth();
