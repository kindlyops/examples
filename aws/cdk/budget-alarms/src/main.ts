import * as budgets from '@aws-cdk/aws-budgets';
import { App, Construct, Stack, StackProps } from '@aws-cdk/core';

export class MyStack extends Stack {
  constructor(scope: Construct, id: string, props: StackProps = {}) {
    super(scope, id, props);

    new budgets.CfnBudget(this, 'MyCfnBudget', {
      budget: {
        budgetType: 'COST',
        timeUnit: 'DAILY',
        budgetLimit: {
          amount: 100,
          unit: 'USD',
        },
        budgetName: 'daily-spend-budget',
        // costFilters: costFilters, this could be a tag, region, account, or service
        costTypes: {
          includeCredit: false,
          includeDiscount: true,
          includeOtherSubscription: true,
          includeRecurring: true,
          includeRefund: true,
          includeSubscription: false,
          includeSupport: false,
          includeTax: false,
          includeUpfront: false,
          useAmortized: false,
          useBlended: false,
        },
        //plannedBudgetLimits: ,

        // when created, the budget will automatically start at the beginning
        // of the current period
        // timePeriod: {
        //   start: 'start',
        //   end: 'end',
        // },
      },
      notificationsWithSubscribers: [
        {
          notification: {
            comparisonOperator: 'GREATER_THAN',
            notificationType: 'ACTUAL',
            threshold: 100,
            thresholdType: 'PERCENTAGE',
          },
          subscribers: [
            {
              address: 'support+budgetexample@kindlyops.com',
              subscriptionType: 'EMAIL',
            },
          ],
        },
      ],
    });
  }
}

// for development, use account/region from cdk cli
const devEnv = {
  account: process.env.CDK_DEFAULT_ACCOUNT,
  region: process.env.CDK_DEFAULT_REGION,
};

const app = new App();

new MyStack(app, 'budget-example-alarms', { env: devEnv });
// new MyStack(app, 'my-stack-prod', { env: prodEnv });

app.synth();
