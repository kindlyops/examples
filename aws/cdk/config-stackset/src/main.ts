import {
  AccountPrincipal,
  Policy,
  PolicyStatement,
  Role,
  ServicePrincipal,
} from '@aws-cdk/aws-iam';
import {
  App,
  CfnStackSet,
  Construct,
  Stack,
  StackProps,
} from '@aws-cdk/core';

export class MyStack extends Stack {
  constructor(scope: Construct, id: string, props: StackProps = {}) {
    super(scope, id, props);

    // set up the special named stack set administration & execution roles
    const stacksetAdminRole = new Role(this, 'configRole', {
      roleName: 'AWSCloudFormationStackSetAdministrationRole',
      assumedBy: new ServicePrincipal('cloudformation.amazonaws.com'),
    });
    stacksetAdminRole.attachInlinePolicy(
      new Policy(this, 'configPolicy', {
        statements: [
          new PolicyStatement({
            actions: ['sts:AssumeRole'],
            resources: ['arn:*:iam::*:role/AWSCloudFormationStackSetExecutionRole'],
          }),
        ],
      }),
    );

    const stacksetExecutionRole = new Role(this, 'executionRole', {
      roleName: 'AWSCloudFormationStackSetExecutionRole',
      assumedBy: new AccountPrincipal(this.account),
    });
    stacksetExecutionRole.attachInlinePolicy(
      new Policy(this, 'executionPolicy', {
        statements: [
          new PolicyStatement({
            actions: [
              'cloudformation:*',
              's3:*',
              'sns:*',
            ],
            resources: [
              '*',
            ],
          }),
        ],
      }),
    );


    const set = new CfnStackSet(this, 'StackSet', {
      stackSetName: 'TopicStackSet',
      permissionModel: 'SELF_MANAGED',
      parameters: [
        { parameterKey: 'loggingRegion', parameterValue: this.region },
      ],
      stackInstancesGroup: [
        {
          // NOTE: the regional STS endpoints must be enabled for every region you
          // want to deploy a stack into.
          regions: [
            'us-west-1',
            'us-west-2',
            'us-east-1',
            'us-east-2',
          ],
          deploymentTargets: {
            accounts: [this.account],
          },
        },
      ],
      templateBody: `
        Parameters:
          loggingRegion:
            Type: String
        Resources:
          Topic:
            Type: AWS::SNS::Topic
            Properties:
              TopicName: !Sub events-to-\${loggingRegion}
      `,
    });

    set.node.addDependency(stacksetAdminRole);
    set.node.addDependency(stacksetExecutionRole);
  }
}

// for development, use account/region from cdk cli
const devEnv = {
  account: process.env.CDK_DEFAULT_ACCOUNT,
  region: process.env.CDK_DEFAULT_REGION,
};

const app = new App();

new MyStack(app, 'kindlyops-stackset-demo-dev', { env: devEnv });
// new MyStack(app, 'my-stack-prod', { env: prodEnv });

app.synth();
