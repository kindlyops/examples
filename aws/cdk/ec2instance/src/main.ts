import { App, Construct, Stack, StackProps } from '@aws-cdk/core';
import * as fs from 'fs';

import * as ec2 from '@aws-cdk/aws-ec2';
import * as iam from '@aws-cdk/aws-iam';
import * as cdk from '@aws-cdk/core';


export class EC2Stack extends Stack {
  constructor(scope: Construct, id: string, props: StackProps = {}) {
    super(scope, id, props);

    const vpc = new ec2.Vpc(this, 'Vpc', { maxAzs: 2 });

    const blockDevices = [
        {
            deviceName: '/dev/xvda',
            volume: ec2.BlockDeviceVolume.ebs(2048, {
                volumeType: ec2.EbsDeviceVolumeType.GP2,
                encrypted: true,
            }),
        },
    ];

    const securityGroup = new ec2.SecurityGroup(this, 'SecurityGroup', {
        description: `EC2 worker security group`,
        vpc: vpc,
    });
    // if you need connections to specific resources, add them here
    // securityGroup.connections.allowTo(props.databases.FOO, ec2.Port.tcp(5432));

    const instance = new ec2.Instance(this, 'Instance', {
        instanceName: 'ALOHA',
        vpc: vpc,
        instanceType: ec2.InstanceType.of(ec2.InstanceClass.C5, ec2.InstanceSize.XLARGE24),
        machineImage: ec2.MachineImage.latestAmazonLinux({
            generation: ec2.AmazonLinuxGeneration.AMAZON_LINUX_2,
        }),
        userData: ec2.UserData.custom(fs.readFileSync('src/cloud-init.sh', 'utf-8')),
        blockDevices: blockDevices,
        securityGroup: securityGroup,
        vpcSubnets: {
            subnetType: ec2.SubnetType.PUBLIC,
        },
    });

    new cdk.CfnOutput(this, 'KindlyExampleInstanceId', { value: instance.instanceId });

    instance.role.addManagedPolicy(
        iam.ManagedPolicy.fromAwsManagedPolicyName('AmazonSSMManagedInstanceCore')
    );

  }
}

// for development, use account/region from cdk cli
const devEnv = {
  account: process.env.CDK_DEFAULT_ACCOUNT,
  region: process.env.CDK_DEFAULT_REGION,
};

const app = new App();

new EC2Stack(app, 'ec2-stack-dev', { env: devEnv });
// new MyStack(app, 'my-stack-prod', { env: prodEnv });

app.synth();