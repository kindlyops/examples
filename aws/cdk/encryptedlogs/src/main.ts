import { App, Construct, Stack, StackProps } from '@aws-cdk/core';
import * as cdk from '@aws-cdk/core';
import * as logs from '@aws-cdk/aws-logs';
import * as kms from '@aws-cdk/aws-kms';
import * as iam from '@aws-cdk/aws-iam';

export class MyStack extends Stack {
  constructor(scope: Construct, id: string, props: StackProps = {}) {
    super(scope, id, props);

    const logKey = new kms.Key(this, 'LogsKinesisKey', {
      enableKeyRotation: true,
      removalPolicy: cdk.RemovalPolicy.DESTROY,
    });

   new logs.LogGroup(this, 'LogGroup', {
      retention: logs.RetentionDays.ONE_WEEK,
      logGroupName: '/kindlyops/examples/kms-encrypted-log-group',
      encryptionKey: logKey,
      removalPolicy: cdk.RemovalPolicy.DESTROY,
    });

    // grant permissions for cloudwatch to use this KMS key
    // https://docs.aws.amazon.com/AmazonCloudWatch/latest/logs/encrypt-log-data-kms.html#cmk-permissions
    logKey.grantEncryptDecrypt(new iam.ServicePrincipal(`logs.${Stack.of(this).region}.amazonaws.com`),);
  }
}

// for development, use account/region from cdk cli
const devEnv = {
  account: process.env.CDK_DEFAULT_ACCOUNT,
  region: process.env.CDK_DEFAULT_REGION,
};

const app = new App();

new MyStack(app, 'my-stack-dev', { env: devEnv });
// new MyStack(app, 'my-stack-prod', { env: prodEnv });

app.synth();