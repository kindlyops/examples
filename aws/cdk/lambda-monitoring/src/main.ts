import * as lambda from '@aws-cdk/aws-lambda';
import * as logs from '@aws-cdk/aws-logs';
import * as cdk from '@aws-cdk/core';
import * as cw from '@aws-cdk/aws-cloudwatch';
import { App, Construct, RemovalPolicy, Stack, StackProps } from '@aws-cdk/core';

export class MyStack extends Stack {
  constructor(scope: Construct, id: string, props: StackProps = {}) {
    super(scope, id, props);

    const lambdaName = `${this.stackName}-ExampleHelloLambda`;
    const logsGroup = new logs.LogGroup(this, 'lambdaLogGroup', {
      logGroupName: `/aws/lambda/${lambdaName}`,
      retention: logs.RetentionDays.THREE_MONTHS,
      removalPolicy: RemovalPolicy.DESTROY,
    });

    const helloLambda = new lambda.Function(this, 'HelloLambda', {
      code: new lambda.InlineCode(
        "exports.handler = async () => { console.log('hello world'); };",
      ),
      functionName: lambdaName,
      handler: 'index.handler',
      runtime: lambda.Runtime.NODEJS_14_X,
    });

    helloLambda.node.addDependency(logsGroup);

    const functionErrors = helloLambda.metricErrors({
      period: cdk.Duration.minutes(1),
    });
    functionErrors.createAlarm(this, 'helloLambdaErrorAlarm', {
      alarmName: `${this.stackName}-helloLambdaErrorAlarm`,
      evaluationPeriods: 1,
      threshold: 1,
      treatMissingData: cw.TreatMissingData.IGNORE,
      comparisonOperator: cw.ComparisonOperator.GREATER_THAN_OR_EQUAL_TO_THRESHOLD,
      });

  }
}

// for development, use account/region from cdk cli
const devEnv = {
  account: process.env.CDK_DEFAULT_ACCOUNT,
  region: process.env.CDK_DEFAULT_REGION,
};

const app = new App();

new MyStack(app, 'example-lambda-monitoring', { env: devEnv });
// new MyStack(app, 'my-stack-prod', { env: prodEnv });

app.synth();
