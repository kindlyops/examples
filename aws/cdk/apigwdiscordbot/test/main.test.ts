import '@aws-cdk/assert/jest';
import { App } from '@aws-cdk/core';
import { SampleDiscordBotStack } from '../src/main';

test('Snapshot', () => {
  const app = new App();
  const stack = new SampleDiscordBotStack(app, 'test');

  expect(stack).not.toHaveResource('AWS::S3::Bucket');
  expect(app.synth().getStackArtifact(stack.artifactId).template).toMatchSnapshot();
});
