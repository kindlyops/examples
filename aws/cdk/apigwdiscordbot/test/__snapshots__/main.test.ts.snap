// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot 1`] = `
Object {
  "Outputs": Object {
    "discordbotendpointdiscordbotapiEndpoint4DA5A53A": Object {
      "Value": Object {
        "Fn::Join": Array [
          "",
          Array [
            "https://",
            Object {
              "Ref": "discordbotendpointdiscordbotapi9F01A7A8",
            },
            ".execute-api.",
            Object {
              "Ref": "AWS::Region",
            },
            ".",
            Object {
              "Ref": "AWS::URLSuffix",
            },
            "/",
            Object {
              "Ref": "discordbotendpointdiscordbotapiDeploymentStageprodF573AD37",
            },
            "/",
          ],
        ],
      },
    },
  },
  "Parameters": Object {
    "AssetParametersad75987e79437d29fb104b1d16300766b19293675cef961db0be4bc68c8ebae0ArtifactHashA6EA5163": Object {
      "Description": "Artifact hash for asset \\"ad75987e79437d29fb104b1d16300766b19293675cef961db0be4bc68c8ebae0\\"",
      "Type": "String",
    },
    "AssetParametersad75987e79437d29fb104b1d16300766b19293675cef961db0be4bc68c8ebae0S3Bucket8BFE70D0": Object {
      "Description": "S3 bucket for asset \\"ad75987e79437d29fb104b1d16300766b19293675cef961db0be4bc68c8ebae0\\"",
      "Type": "String",
    },
    "AssetParametersad75987e79437d29fb104b1d16300766b19293675cef961db0be4bc68c8ebae0S3VersionKeyF480D8E8": Object {
      "Description": "S3 key for asset version \\"ad75987e79437d29fb104b1d16300766b19293675cef961db0be4bc68c8ebae0\\"",
      "Type": "String",
    },
    "AssetParametersffcf1a77c08a501557af6b7bc169a2e9a9a4b5614ab3963432f0c7c2ec7bbe63ArtifactHash48BD6748": Object {
      "Description": "Artifact hash for asset \\"ffcf1a77c08a501557af6b7bc169a2e9a9a4b5614ab3963432f0c7c2ec7bbe63\\"",
      "Type": "String",
    },
    "AssetParametersffcf1a77c08a501557af6b7bc169a2e9a9a4b5614ab3963432f0c7c2ec7bbe63S3Bucket81FF9F34": Object {
      "Description": "S3 bucket for asset \\"ffcf1a77c08a501557af6b7bc169a2e9a9a4b5614ab3963432f0c7c2ec7bbe63\\"",
      "Type": "String",
    },
    "AssetParametersffcf1a77c08a501557af6b7bc169a2e9a9a4b5614ab3963432f0c7c2ec7bbe63S3VersionKey55C1E5A4": Object {
      "Description": "S3 key for asset version \\"ffcf1a77c08a501557af6b7bc169a2e9a9a4b5614ab3963432f0c7c2ec7bbe63\\"",
      "Type": "String",
    },
  },
  "Resources": Object {
    "discordbotendpointdiscordbotapi9F01A7A8": Object {
      "Properties": Object {
        "Name": "discord-bot-api",
      },
      "Type": "AWS::ApiGateway::RestApi",
    },
    "discordbotendpointdiscordbotapiAccount4AD25378": Object {
      "DependsOn": Array [
        "discordbotendpointdiscordbotapi9F01A7A8",
      ],
      "Properties": Object {
        "CloudWatchRoleArn": Object {
          "Fn::GetAtt": Array [
            "discordbotendpointdiscordbotapiCloudWatchRole0681439A",
            "Arn",
          ],
        },
      },
      "Type": "AWS::ApiGateway::Account",
    },
    "discordbotendpointdiscordbotapiCloudWatchRole0681439A": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "apigateway.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AmazonAPIGatewayPushToCloudWatchLogs",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "discordbotendpointdiscordbotapiDeployment10919DFF8b68aeda3e269b11ce70267b4e451883": Object {
      "DependsOn": Array [
        "discordbotendpointdiscordbotapivalidator2F5873B0",
        "discordbotendpointdiscordbotapieventOPTIONS94181C76",
        "discordbotendpointdiscordbotapieventPOSTB34843CC",
        "discordbotendpointdiscordbotapievent261096F7",
        "discordbotendpointdiscordbotapiOPTIONS7D908FD8",
      ],
      "Properties": Object {
        "Description": "Automatically created by the RestApi construct",
        "RestApiId": Object {
          "Ref": "discordbotendpointdiscordbotapi9F01A7A8",
        },
      },
      "Type": "AWS::ApiGateway::Deployment",
    },
    "discordbotendpointdiscordbotapiDeploymentStageprodF573AD37": Object {
      "Properties": Object {
        "DeploymentId": Object {
          "Ref": "discordbotendpointdiscordbotapiDeployment10919DFF8b68aeda3e269b11ce70267b4e451883",
        },
        "RestApiId": Object {
          "Ref": "discordbotendpointdiscordbotapi9F01A7A8",
        },
        "StageName": "prod",
      },
      "Type": "AWS::ApiGateway::Stage",
    },
    "discordbotendpointdiscordbotapiOPTIONS7D908FD8": Object {
      "Properties": Object {
        "AuthorizationType": "NONE",
        "HttpMethod": "OPTIONS",
        "Integration": Object {
          "IntegrationResponses": Array [
            Object {
              "ResponseParameters": Object {
                "method.response.header.Access-Control-Allow-Headers": "'Content-Type,X-Amz-Date,Authorization,X-Api-Key,X-Amz-Security-Token,X-Amz-User-Agent'",
                "method.response.header.Access-Control-Allow-Methods": "'OPTIONS,GET,PUT,POST,DELETE,PATCH,HEAD'",
                "method.response.header.Access-Control-Allow-Origin": "'*'",
              },
              "StatusCode": "204",
            },
          ],
          "RequestTemplates": Object {
            "application/json": "{ statusCode: 200 }",
          },
          "Type": "MOCK",
        },
        "MethodResponses": Array [
          Object {
            "ResponseParameters": Object {
              "method.response.header.Access-Control-Allow-Headers": true,
              "method.response.header.Access-Control-Allow-Methods": true,
              "method.response.header.Access-Control-Allow-Origin": true,
            },
            "StatusCode": "204",
          },
        ],
        "ResourceId": Object {
          "Fn::GetAtt": Array [
            "discordbotendpointdiscordbotapi9F01A7A8",
            "RootResourceId",
          ],
        },
        "RestApiId": Object {
          "Ref": "discordbotendpointdiscordbotapi9F01A7A8",
        },
      },
      "Type": "AWS::ApiGateway::Method",
    },
    "discordbotendpointdiscordbotapievent261096F7": Object {
      "Properties": Object {
        "ParentId": Object {
          "Fn::GetAtt": Array [
            "discordbotendpointdiscordbotapi9F01A7A8",
            "RootResourceId",
          ],
        },
        "PathPart": "event",
        "RestApiId": Object {
          "Ref": "discordbotendpointdiscordbotapi9F01A7A8",
        },
      },
      "Type": "AWS::ApiGateway::Resource",
    },
    "discordbotendpointdiscordbotapieventOPTIONS94181C76": Object {
      "Properties": Object {
        "AuthorizationType": "NONE",
        "HttpMethod": "OPTIONS",
        "Integration": Object {
          "IntegrationResponses": Array [
            Object {
              "ResponseParameters": Object {
                "method.response.header.Access-Control-Allow-Headers": "'Content-Type,X-Amz-Date,Authorization,X-Api-Key,X-Amz-Security-Token,X-Amz-User-Agent'",
                "method.response.header.Access-Control-Allow-Methods": "'OPTIONS,GET,PUT,POST,DELETE,PATCH,HEAD'",
                "method.response.header.Access-Control-Allow-Origin": "'*'",
              },
              "StatusCode": "204",
            },
          ],
          "RequestTemplates": Object {
            "application/json": "{ statusCode: 200 }",
          },
          "Type": "MOCK",
        },
        "MethodResponses": Array [
          Object {
            "ResponseParameters": Object {
              "method.response.header.Access-Control-Allow-Headers": true,
              "method.response.header.Access-Control-Allow-Methods": true,
              "method.response.header.Access-Control-Allow-Origin": true,
            },
            "StatusCode": "204",
          },
        ],
        "ResourceId": Object {
          "Ref": "discordbotendpointdiscordbotapievent261096F7",
        },
        "RestApiId": Object {
          "Ref": "discordbotendpointdiscordbotapi9F01A7A8",
        },
      },
      "Type": "AWS::ApiGateway::Method",
    },
    "discordbotendpointdiscordbotapieventPOSTApiPermissionTesttestdiscordbotendpointdiscordbotapiE3ED2527POSTevent61396A22": Object {
      "Properties": Object {
        "Action": "lambda:InvokeFunction",
        "FunctionName": Object {
          "Fn::GetAtt": Array [
            "discordbotendpointdiscordbotlambdaEDF529AC",
            "Arn",
          ],
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": Object {
          "Fn::Join": Array [
            "",
            Array [
              "arn:",
              Object {
                "Ref": "AWS::Partition",
              },
              ":execute-api:",
              Object {
                "Ref": "AWS::Region",
              },
              ":",
              Object {
                "Ref": "AWS::AccountId",
              },
              ":",
              Object {
                "Ref": "discordbotendpointdiscordbotapi9F01A7A8",
              },
              "/test-invoke-stage/POST/event",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "discordbotendpointdiscordbotapieventPOSTApiPermissiontestdiscordbotendpointdiscordbotapiE3ED2527POSTeventD44CEF01": Object {
      "Properties": Object {
        "Action": "lambda:InvokeFunction",
        "FunctionName": Object {
          "Fn::GetAtt": Array [
            "discordbotendpointdiscordbotlambdaEDF529AC",
            "Arn",
          ],
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": Object {
          "Fn::Join": Array [
            "",
            Array [
              "arn:",
              Object {
                "Ref": "AWS::Partition",
              },
              ":execute-api:",
              Object {
                "Ref": "AWS::Region",
              },
              ":",
              Object {
                "Ref": "AWS::AccountId",
              },
              ":",
              Object {
                "Ref": "discordbotendpointdiscordbotapi9F01A7A8",
              },
              "/",
              Object {
                "Ref": "discordbotendpointdiscordbotapiDeploymentStageprodF573AD37",
              },
              "/POST/event",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "discordbotendpointdiscordbotapieventPOSTB34843CC": Object {
      "Properties": Object {
        "AuthorizationType": "NONE",
        "HttpMethod": "POST",
        "Integration": Object {
          "IntegrationHttpMethod": "POST",
          "IntegrationResponses": Array [
            Object {
              "StatusCode": "200",
            },
            Object {
              "ResponseTemplates": Object {
                "application/json": "invalid request signature",
              },
              "SelectionPattern": ".*[UNAUTHORIZED].*",
              "StatusCode": "401",
            },
          ],
          "RequestTemplates": Object {
            "application/json": "{
              \\"timestamp\\": \\"$input.params('x-signature-timestamp')\\",
              \\"signature\\": \\"$input.params('x-signature-ed25519')\\",
              \\"jsonBody\\" : $input.json('$')
            }",
          },
          "Type": "AWS",
          "Uri": Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":apigateway:",
                Object {
                  "Ref": "AWS::Region",
                },
                ":lambda:path/2015-03-31/functions/",
                Object {
                  "Fn::GetAtt": Array [
                    "discordbotendpointdiscordbotlambdaEDF529AC",
                    "Arn",
                  ],
                },
                "/invocations",
              ],
            ],
          },
        },
        "MethodResponses": Array [
          Object {
            "StatusCode": "200",
          },
          Object {
            "StatusCode": "401",
          },
        ],
        "RequestValidatorId": Object {
          "Ref": "discordbotendpointdiscordbotapivalidator2F5873B0",
        },
        "ResourceId": Object {
          "Ref": "discordbotendpointdiscordbotapievent261096F7",
        },
        "RestApiId": Object {
          "Ref": "discordbotendpointdiscordbotapi9F01A7A8",
        },
      },
      "Type": "AWS::ApiGateway::Method",
    },
    "discordbotendpointdiscordbotapivalidator2F5873B0": Object {
      "Properties": Object {
        "RestApiId": Object {
          "Ref": "discordbotendpointdiscordbotapi9F01A7A8",
        },
        "ValidateRequestBody": true,
        "ValidateRequestParameters": true,
      },
      "Type": "AWS::ApiGateway::RequestValidator",
    },
    "discordbotendpointdiscordbotlambdaEDF529AC": Object {
      "DependsOn": Array [
        "discordbotendpointdiscordbotlambdaServiceRoleDefaultPolicy23EF68EB",
        "discordbotendpointdiscordbotlambdaServiceRoleE1BA8F0E",
      ],
      "Properties": Object {
        "Code": Object {
          "S3Bucket": Object {
            "Ref": "AssetParametersffcf1a77c08a501557af6b7bc169a2e9a9a4b5614ab3963432f0c7c2ec7bbe63S3Bucket81FF9F34",
          },
          "S3Key": Object {
            "Fn::Join": Array [
              "",
              Array [
                Object {
                  "Fn::Select": Array [
                    0,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParametersffcf1a77c08a501557af6b7bc169a2e9a9a4b5614ab3963432f0c7c2ec7bbe63S3VersionKey55C1E5A4",
                        },
                      ],
                    },
                  ],
                },
                Object {
                  "Fn::Select": Array [
                    1,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParametersffcf1a77c08a501557af6b7bc169a2e9a9a4b5614ab3963432f0c7c2ec7bbe63S3VersionKey55C1E5A4",
                        },
                      ],
                    },
                  ],
                },
              ],
            ],
          },
        },
        "Environment": Object {
          "Variables": Object {
            "AWS_NODEJS_CONNECTION_REUSE_ENABLED": "1",
            "COMMAND_LAMBDA_ARN": Object {
              "Fn::GetAtt": Array [
                "discordcommandslambda383AD657",
                "Arn",
              ],
            },
            "DISCORD_BOT_API_KEY_NAME": Object {
              "Fn::Select": Array [
                6,
                Object {
                  "Fn::Split": Array [
                    ":",
                    Object {
                      "Ref": "discordbotendpointoculusstartdiscordapikey9E0BAD85",
                    },
                  ],
                },
              ],
            },
          },
        },
        "Handler": "index.handler",
        "Role": Object {
          "Fn::GetAtt": Array [
            "discordbotendpointdiscordbotlambdaServiceRoleE1BA8F0E",
            "Arn",
          ],
        },
        "Runtime": "nodejs14.x",
        "Timeout": 3,
      },
      "Type": "AWS::Lambda::Function",
    },
    "discordbotendpointdiscordbotlambdaServiceRoleDefaultPolicy23EF68EB": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "secretsmanager:GetSecretValue",
                "secretsmanager:DescribeSecret",
              ],
              "Effect": "Allow",
              "Resource": Object {
                "Ref": "discordbotendpointoculusstartdiscordapikey9E0BAD85",
              },
            },
            Object {
              "Action": "lambda:InvokeFunction",
              "Effect": "Allow",
              "Resource": Object {
                "Fn::GetAtt": Array [
                  "discordcommandslambda383AD657",
                  "Arn",
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "discordbotendpointdiscordbotlambdaServiceRoleDefaultPolicy23EF68EB",
        "Roles": Array [
          Object {
            "Ref": "discordbotendpointdiscordbotlambdaServiceRoleE1BA8F0E",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "discordbotendpointdiscordbotlambdaServiceRoleE1BA8F0E": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "discordbotendpointoculusstartdiscordapikey9E0BAD85": Object {
      "DeletionPolicy": "Delete",
      "Properties": Object {
        "GenerateSecretString": Object {},
      },
      "Type": "AWS::SecretsManager::Secret",
      "UpdateReplacePolicy": "Delete",
    },
    "discordcommandslambda383AD657": Object {
      "DependsOn": Array [
        "discordcommandslambdaServiceRoleDefaultPolicy09F93C78",
        "discordcommandslambdaServiceRole1D2A4A42",
      ],
      "Properties": Object {
        "Code": Object {
          "S3Bucket": Object {
            "Ref": "AssetParametersad75987e79437d29fb104b1d16300766b19293675cef961db0be4bc68c8ebae0S3Bucket8BFE70D0",
          },
          "S3Key": Object {
            "Fn::Join": Array [
              "",
              Array [
                Object {
                  "Fn::Select": Array [
                    0,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParametersad75987e79437d29fb104b1d16300766b19293675cef961db0be4bc68c8ebae0S3VersionKeyF480D8E8",
                        },
                      ],
                    },
                  ],
                },
                Object {
                  "Fn::Select": Array [
                    1,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParametersad75987e79437d29fb104b1d16300766b19293675cef961db0be4bc68c8ebae0S3VersionKeyF480D8E8",
                        },
                      ],
                    },
                  ],
                },
              ],
            ],
          },
        },
        "Environment": Object {
          "Variables": Object {
            "AWS_NODEJS_CONNECTION_REUSE_ENABLED": "1",
            "DISCORD_BOT_API_KEY_NAME": Object {
              "Fn::Select": Array [
                6,
                Object {
                  "Fn::Split": Array [
                    ":",
                    Object {
                      "Ref": "discordbotendpointoculusstartdiscordapikey9E0BAD85",
                    },
                  ],
                },
              ],
            },
          },
        },
        "Handler": "index.handler",
        "Role": Object {
          "Fn::GetAtt": Array [
            "discordcommandslambdaServiceRole1D2A4A42",
            "Arn",
          ],
        },
        "Runtime": "nodejs14.x",
        "Timeout": 60,
      },
      "Type": "AWS::Lambda::Function",
    },
    "discordcommandslambdaServiceRole1D2A4A42": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "discordcommandslambdaServiceRoleDefaultPolicy09F93C78": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "secretsmanager:GetSecretValue",
                "secretsmanager:DescribeSecret",
              ],
              "Effect": "Allow",
              "Resource": Object {
                "Ref": "discordbotendpointoculusstartdiscordapikey9E0BAD85",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "discordcommandslambdaServiceRoleDefaultPolicy09F93C78",
        "Roles": Array [
          Object {
            "Ref": "discordcommandslambdaServiceRole1D2A4A42",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
  },
}
`;
